samples/tools/autogenbench/scenarios/GAIA/Scripts/demystifying-telemetry-sqlite.ipynb

```diff
@@ -0,0 +1,700 @@
+    "    elif event[\"event_name\"] == \"request_to_oai_cc\":\n",
+    "        for message in event[\"request\"][\"messages\"]:\n",
+    "            if isinstance(message[\"content\"], list):\n",
+    "                return subcontent_pass(message[\"content\"])\n",
+    "    elif event[\"event_name\"] == \"reply_func_executed\":\n",
+    "        json_state = event[\"json_state\"]\n",
+    "        return subcontent_pass(json_state[\"reply\"][\"content\"])\n",
+    "    else:\n",
+    "        raise Exception(f\"image detected in event type {event['event_name']} but not loaded.\")\n",
+    "\n",
+    "    return image_url\n",
+    "\n",
+    "\n",
+    "# load table events and do some processing\n",
+    "def db_events_to_list(conn):\n",
+    "    c = conn.cursor()\n",
+    "    c.execute(\"SELECT * FROM events\")\n",
+    "    rows = c.fetchall()\n",
+    "    colnames = c.description\n",
+    "\n",
+    "    final = [dict(zip([col[0] for col in colnames], row)) for row in rows]\n",
+    "    for e in final:\n",
+    "        e[\"image_content\"] = \"data:image/png;base64\" in e[\"json_state\"]\n",
+    "        e[\"json_state\"] = json.loads(e[\"json_state\"])\n",
+    "        if e[\"image_content\"]:\n",
+    "            e[\"image\"] = get_first_image_from_event(e)\n",
+    "        e[\"timestamp\"] = datetime.strptime(e[\"timestamp\"], \"%Y-%m-%d %H:%M:%S.%f\")\n",
+    "    return final\n",
+    "\n",
+    "\n",
+    "# load table chat_completions and do some processing\n",
+    "def db_cc_to_list(conn):\n",
+    "    c = conn.cursor()\n",
+    "    c.execute(\"SELECT * FROM chat_completions\")\n",
+    "    rows = c.fetchall()\n",
+    "    colnames = c.description\n",
+    "\n",
+    "    final = [dict(zip([col[0] for col in colnames], row)) for row in rows]\n",
+    "    for e in final:\n",
+    "        e[\"image_content\"] = \"data:image/png;base64\" in e[\"request\"]\n",
+    "        e[\"request\"] = json.loads(e[\"request\"])\n",
+    "        e[\"timestamp\"] = datetime.strptime(e[\"start_time\"], \"%Y-%m-%d %H:%M:%S.%f\")\n",
+    "        e[\"start_time\"] = datetime.strptime(e[\"start_time\"], \"%Y-%m-%d %H:%M:%S.%f\")\n",
+    "        e[\"end_time\"] = datetime.strptime(e[\"end_time\"], \"%Y-%m-%d %H:%M:%S.%f\")\n",
+    "        e[\"total_seconds\"] = (e[\"end_time\"] - e[\"start_time\"]).seconds\n",
+    "        e[\"event_name\"] = \"request_to_oai_cc\"\n",
+    "        e[\"source_name\"] = \"chat_completions_table\"\n",
+    "        if e[\"image_content\"]:\n",
+    "            e[\"image\"] = get_first_image_from_event(e)\n",
+    "    return final"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Brief explanation\n",
+    "\n",
+    "The logs collected by scenario.py in https://github.com/microsoft/autogen/blob/ct_webarena/samples/tools/autogenbench/scenarios/WebArena/Templates/TwoAgents/scenario.py#L50\n",
+    "\n",
```

Comment:
not sure if this line is out of date, line 50 of scenario.py is pointing to the task not log_event? Maybe just remove the link?

Reply:
outdated, yes